### ISSUE TITLE
Personnalisation du thème CSS

### ISSUE THREAD

**Issue by akarzim** [betagouv/demarches-simplifiees.fr#6879]
# Résumé

Cette proposition permet de modifier le thème CSS d'une instance Démarches Simplifiées à l'aide d'une simple feuille de style personnalisée qui peut être servie statiquement.

mots-clés : couleur, css, thème

# Actuellement

Les couleurs qu'arbore Démarches Simplifiées sont définies à l'aide de variables SCSS. Ce qui empêche de les surcharger via une feuille de style qui serait servie statiquement.

Or, si l'on veut surcharger les couleurs sans avoir à pousser sur le dépôt Git des feuilles de style personnalisées, il nous faut pouvoir les servir statiquement.

# Comportement attendu

Une instance Démarches Simplifiées voulant adapter le thème par défaut pour une meilleure intégration à sa charge graphique, doit pouvoir fournir une feuille de style personnalisée pour ce faire.

# Proposition

Utilisation de l'instruction CSS `var()` faisant appel à des variable CSS pour définir les couleurs SCSS.

```scss
:root {
	--blue-france-500: #000091;
}

$blue-france-500: var(--blue-france-500);
```

Ainsi, les variables CSS pourront être surchargées à travers une feuille de style personnalisée dont le chemin sera connu par le moyen d'une variable d'environnement `EXTERNAL_STYLESHEET`. La feuille de style personnalisée sera chargée si cette variable est définie.

--- 

voir : PR à venir / @adullact & @synbioz

> - L'ADULLACT a mandaté le prestataire @synbioz pour développer plusieurs fonctionnalités (tickets et PR à venir).
> - C'est dans ce cadre que @synbioz nous propose certains correctifs annexes comme celui-ci.
> - Si c'est nécessaire, @akarzim et @jobygoude de @synbioz pourront interagir avec l'équipe @betagouv sur ce ticket et sur la PR (répondre aux commentaires, pousser des commits…).

---

`trackingAdullactContrib`

**Comment by akarzim** [comment]
@tchak tel que proposé, à l'aide de variables CSS, apparemment non.

<img width="1018" alt="image" src="https://user-images.githubusercontent.com/130560/152311271-155d0ff7-ccf5-4e46-b42f-8e7cbbd5a2f1.png">

> source: https://developer.mozilla.org/en-US/docs/Web/CSS/var()


Ceci étant, la part de marché d'IE 11 en France en 2021 est de seulement 1,06%. Et même les navigateurs encore plus minoritaires tels que Samsung Internet 12 & 13 ou Edge 18 supportent les variables CSS. IE 11 fait figure d'exception.

![image](https://user-images.githubusercontent.com/130560/152311210-6239d18f-df1a-4d8d-8f90-ceb9a7fa3186.png)
> source: https://www.leptidigital.fr/webmarketing/parts-de-marche-navigateurs-web-10814/

Par ailleurs, le support d'IE 11 — sorti en octobre 2013 — prendra fin au 15 juin 2022.

> Internet Explorer (IE) 11 est la dernière version majeure d’Internet Explorer. À compter du 15 juin 2022, l’application de bureau Internet Explorer 11 ne sera plus prise en charge sur certaines versions de Windows 10*.
>
> Les clients sont encouragés à passer à [Microsoft Edge](https://www.microsoft.com/edge/business/download?form=MO12H4&OCID=MO12H4) qui prend en charge les sites Web et les applications hérités et modernes.

> source: https://docs.microsoft.com/fr-fr/lifecycle/announcements/internet-explorer-11-end-of-support

Questions : 
- Doit-on se priver de la possibilité d'adapter le thème CSS aux couleurs de l'organisation qui met une instance DS à disposition dans le seul but d'assurer cette compatibilité ?
- Quels sont les enjeux de cette compatibilité ?
- Quelles alternatives s'offrent à nous pour apporter de la souplesse au niveau des couleurs dominantes de l'application ?

Précision : il s'agit là d'un besoin exprimé par l'Adullact et plusieurs collectivités.

**Comment by tchak** [comment]
Je suis d'accord avec tous tes arguments. Nous sommes en train d'avoir une discussion interne sur le sujet. Notre première estimation est que nous avons 4% d’usagers IE11. Nous allons faire un travail pour affiner et vérifier ce chiffre. S’il se confirme, on ne pourra pas renoncer au support de IE11.

Je suis d’accord avec l’objectif de cette PR. De notre côté, nous avons un sujet pour passer sur le “design sytem de l’état” qui va nous demander de gros changements sur l’architecture de notre css. Une fois que nous avons pris la décision sur IE il faudra qu’on se penche sur les solutions possibles.

**Comment by dzc34** [comment]
De notre côté, la volumétrie n'est pas assez encore assez importante pour en tirer toutes les conclusions. Sur 1 an, c'est 1% de visiteurs et plutôt des visiteurs appartenant à des collectivités (pages administrateur, pages instructeur et pages usager sur des démarches ADULLACT pour les collectivités).

**Comment by tchak** [comment]
Oui, on pensait le faire. Mais au vu des chiffres et des dernières discussions dans l'équipe, on ne l'appliquera pas. On va au minimum changer le message dans l'application.
